dev/bots/entry.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+# the bot entry-point - delegate to the correct script
+
+if [ "$BUILD_DOCS" = "true" ]; then
+  # generate docs
+  dev/bots/docs.sh
```

Comment:
Still doing a bit of surgery to this PR, but in the end it'll run three bots for each commit, with different env vars set up for each bot. This will split the work for testing and doc gen into two separate bots, to make each run faster, and add a new bot which will give us early warning of issues with the next dart sdk roll.